src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataControl.xaml.cs

```diff
@@ -2,76 +2,67 @@
 {
 
     /// <summary>
     /// Interaction logic for PackageMetadataControl.xaml
     /// </summary>
-    public partial class PackageMetadataControl : UserControl
+    public partial class PackageMetadataControl : UserControl, IDisposable
     {
         public ReadMePreviewViewModel ReadMePreviewViewModel { get; set; }
 
+        private bool _initialVisibilitySet = false;
+
         public PackageMetadataControl()
         {
             InitializeComponent();
             Visibility = Visibility.Collapsed;
             DataContextChanged += PackageMetadataControl_DataContextChanged;
         }
 
-#pragma warning disable VSTHRD100 // Avoid async void methods
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _packageMetadataReadMeControl?.Dispose();
+            }
+        }
+
         private void PackageMetadataControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
-#pragma warning restore VSTHRD100 // Avoid async void methods
         {
             if (DataContext is DetailControlModel detailControlModel)
             {
-                detailControlModel.PropertyChanged += PackageMetadataControl_DataContext_PropertyChanged;
-                if (string.IsNullOrWhiteSpace(detailControlModel.PackagePath))
-                {
-                    tabReadMe.Visibility = Visibility.Collapsed;
-                    tabPackageDetails.IsSelected = true;
-                }
-                else
-                {
-                    tabReadMe.Visibility = Visibility.Visible;
-                }
                 Visibility = Visibility.Visible;
             }
             else
             {
-                tabReadMe.Visibility = Visibility.Collapsed;
                 Visibility = Visibility.Collapsed;
             }
         }
 
-        private void PackageMetadataControl_DataContext_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        private void TabReadMe_IsVisibleChanged(object sender, DependencyPropertyChangedEventArgs e)
         {
-            if (sender is DetailControlModel detailControlModel)
+            if (!_initialVisibilitySet && tabReadMe.IsVisible)
             {
-                if (e.PropertyName == nameof(detailControlModel.IsReadMeAvailable))
-                {
-                    if (!detailControlModel.IsReadMeAvailable)
-                    {
-                        tabReadMe.Visibility = Visibility.Collapsed;
-                        tabPackageDetails.IsSelected = true;
-                    }
-                    else
-                    {
-                        tabReadMe.Visibility = Visibility.Visible;
-                    }
-
-                }
-
+                tabReadMe.IsSelected = true;
+                tabPackageDetails.IsSelected = false;
```

Comment:
Do we have to set all other tabs as IsSelected = false? Is it sufficient to set the tab we want selected to true? I'm not certain, but I believe that's how the tab control works. You'd need to test to be sure, but I don't believe it allows for more than one tab to be selected at a time.

Reply:
yup looks like it's not needed